docker/bootstrap/Dockerfile.mysql84

```diff
@@ -0,0 +1,23 @@
+ARG bootstrap_version
+ARG image="vitess/bootstrap:${bootstrap_version}-common"
+
+FROM --platform=linux/amd64 "${image}"
+
+USER root
+
+# Install MySQL 8.4
+RUN for i in $(seq 1 10); do apt-key adv --no-tty --recv-keys --keyserver keyserver.ubuntu.com 8C718D3B5072E1F5 && break; done && \
+    for i in $(seq 1 10); do apt-key adv --no-tty --recv-keys --keyserver keyserver.ubuntu.com A8D3785C && break; done && \
+    add-apt-repository 'deb http://repo.mysql.com/apt/debian/ bullseye mysql-8.4' && \
```

Comment:
@frouioui Do we have any blockers on upgrading to bookworm? We should get things onto that since `bullseye` is already in the LTS phase.

Reply:
I don't think we would have any blocker. As discussed offline, we should keep the bullseye builds, move to bookworm as the default, and bump MySQL to 8.0.40 so we can use bookworm. 

https://github.com/vitessio/vitess/pull/17539